### DISCUSSION TITLE
Sonoff THR320D with THS01

### DISCUSSION THREAD

**Discussion by desiue** [arendst/Tasmota#21719] [Q&A]
Hello everyone,

i have an new one Sonoff THR320D with THS01 Sensor but have several Problems.

1. it reacts very slowly, sometimes ist hanging, even though it has an esp32 chip in it. the old Sonof TH10/16 with the ESP8266 are faster.
2. Sometimes only null values are shown in the web display. But on the device Display the right values.
3. Sometimes the display does not come on or it takes a few minutes for it to come on.  Sometimes the display switches off.
Mostly restart helps so that the values are shown again on the web interface and on the display, but not always.
Sometimes the values are shown on the display but not on the web interfcae or vice versa. It's maddening.

All in all, it looks as if something is wrong. I have some Sonoff TH10/16 and D1 Mini with Tasmota that work very well.

I have flashed the firmware several times with "ESP-Flasher" and via the web interface.  Version 14.1.0 
http://ota.tasmota.com/tasmota32/release/tasmota32.bin

I then selected "auto-configuration" "Sonoff THR320D with THS01" under "configuration".
Then "Configuration" -> "Configure Module" IO GPIO25 "SI7021". Because THS01 is the SI7021.

My other attempt was to use a template.  https://templates.blakadder.com/sonoff_THR316D.html

{"NAME":"Sonoff THR316D","GPIO":[32,0,0,0,225,9280,0,0,0,321,0,576,320,9184,9216,0,0,224,0,9248,0,1,0,3840,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}

and there was another one from this site https://bangertech.de/sonoff-th-elite-x-tasmota/

{"NAME":"Sonoff THR320D","GPIO":[32,0,0,0,226,9280,0,0,0,321,0,576,320,9184,9216,9312,0,0,9313,9248,0,1,0,3840,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}

But that didn't help either. 
I have set Latitude, Longitude, TimeDST, TimeSTD and TimeZone.
I also tested the SetOption74  to switch it on and off, didn't help either.

I'm slowly running out of ideas.

Does anyone use this Sonoff hardware? Do you have any tips? 

![Screenshot 2024-07-02 154224](https://github.com/arendst/Tasmota/assets/77403450/d3a3910b-927d-41a3-b3d9-b7df160b7d34)

My next Idea was to set the Web log level to 0

```

17:02:30.228 CFG: Saved, Count 23, Bytes 4096
17:02:30.735 HTP: Main Menu
17:02:46.532 DHT: Pin25 cycles (0/80)  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ..
17:02:46.532 DHT: Pin25 timeout waiting for pulse 0
17:02:50.562 DHT: Pin25 cycles (0/80)  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ..
17:02:50.562 DHT: Pin25 timeout waiting for pulse 0
17:02:54.542 DHT: Pin25 cycles (0/80)  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ..
17:02:54.542 DHT: Pin25 timeout waiting for pulse 0
17:02:55.947 WIF: Checking connection...
17:02:57.348 HTP: Main Menu
17:02:58.573 DHT: Pin25 cycles (0/80)  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ..
17:02:58.574 DHT: Pin25 timeout waiting for pulse 0
17:03:01.878 HTP: Tools
17:03:02.542 DHT: Pin25 cycles (0/80)  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ..
17:03:02.543 DHT: Pin25 timeout waiting for pulse 0
17:03:03.697 HTP: Console
17:03:06.535 DHT: Pin25 cycles (0/80)  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ..
17:03:06.536 DHT: Pin25 timeout waiting for pulse 0
17:03:10.565 DHT: Pin25 cycles (0/80)  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ..


```

With this messages i found this thread https://github.com/arendst/Tasmota/issues/17944

so ist seems  DhtDelay 500,40  helped a little bit....
DhtDelay will be resetet after restart. so mybe a Rule is an option

Rule1 on system#init do DhtDelay 500,40 endon
Rule1 1

It still feels a bit sluggish and the wifi reception is also poor. All in all, not a good buy.

**Comment by arendst** [comment]
I have good results:

```
17:23:14.873 RUL: SYSTEM#BOOT performs 'dhtdelay 450,40'
17:23:14.874 SRC: Rule
17:23:14.875 CMD: Grp 0, Cmd 'DHTDELAY', Idx 1, Len 6, Pld 450, Data '450,40'
17:23:14.891 MQT: stat/thr316d/RESULT = {"DhtDelay1":[450,40]}
17:23:16.562 HTP: Main Menu
17:23:16.625 WIF: IPv6 Global fd84:7996:2cbf:1349:7a21:84ff:feba:ae74
17:23:16.932 QPC: Reset
17:23:18.922 APP: Boot Count 360
17:23:18.930 MQT: tele/thr316d/STATE = {"Time":"2024-07-02T17:23:18","Uptime":"0T00:00:09","UptimeSec":9,"Heap":157,"SleepMode":"Dynamic","Sleep":0,"LoadAvg":321,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":31},"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"indebuurt_IoT","BSSId":"18:E8:29:CA:17:C1","Channel":11,"Mode":"HT20","RSSI":100,"Signal":-41,"LinkCount":1,"Downtime":"0T00:00:03"}}
17:23:18.968 MQT: tele/thr316d/SENSOR = {"Time":"2024-07-02T17:23:18","SI7021":{"Temperature":22.3,"Humidity":59.2,"DewPoint":13.9,"HeatIndex":22.1},"TempUnit":"C"}
17:23:19.478 CFG: Saved, Count 1118, Bytes 4096
17:23:23.289 HTP: Tools
17:23:25.005 HTP: Console
17:23:57.204 HTP: Tools
17:23:58.994 HTP: Main Menu
17:24:18.927 MQT: tele/thr316d/STATE = {"Time":"2024-07-02T17:24:18","Uptime":"0T00:01:09","UptimeSec":69,"Heap":156,"SleepMode":"Dynamic","Sleep":0,"LoadAvg":484,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":31},"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"indebuurt_IoT","BSSId":"18:E8:29:CA:17:C1","Channel":11,"Mode":"HT20","RSSI":100,"Signal":-40,"LinkCount":1,"Downtime":"0T00:00:03"}}
17:24:18.965 MQT: tele/thr316d/SENSOR = {"Time":"2024-07-02T17:24:18","SI7021":{"Temperature":22.3,"Humidity":59.2,"DewPoint":13.9,"HeatIndex":22.1},"TempUnit":"C"}
17:24:33.798 HTP: Tools
17:24:35.609 HTP: Console
17:24:43.144 CMD: weblog 4
17:24:43.145 SRC: WebConsole from 192.168.2.1
17:24:43.147 CMD: Grp 0, Cmd 'WEBLOG', Idx 1, Len 1, Pld 4, Data '4'
17:24:43.151 MQT: stat/thr316d/RESULT = {"WebLog":4}
17:24:43.747 CFG: Saved, Count 1119, Bytes 4096
17:24:44.929 DHT: Pin25 cycles (80/80)  53 31 56 31 56 32 55 32 56 31 56 31 56 98 55 39 56 31 56 98 ..
17:24:44.930 DHT: Pin25 read 025000DF31
17:24:48.931 DHT: Pin25 cycles (80/80)  53 31 56 31 56 31 56 32 56 31 56 31 56 97 56 39 56 31 56 98 ..
17:24:48.932 DHT: Pin25 read 024F00DF30
17:24:52.931 DHT: Pin25 cycles (80/80)  53 31 56 31 56 31 56 32 56 31 56 31 56 97 56 39 56 31 56 98 ..
17:24:52.932 DHT: Pin25 read 024F00DF30
17:24:53.811 WIF: Checking connection...
17:24:56.929 DHT: Pin25 cycles (80/80)  53 31 56 31 56 31 56 31 56 32 56 31 56 97 56 39 56 31 56 97 ..
17:24:56.930 DHT: Pin25 read 024F00DF30
17:25:00.930 DHT: Pin25 cycles (80/80)  53 31 56 31 56 32 55 32 56 31 56 31 56 98 55 39 56 31 56 98 ..
17:25:00.931 DHT: Pin25 read 024F00DF30
```

**Reply by VanillaFord** [reply]
I can confirm the problem described by @desiue 

Running tasmota32.bin on the THR320D device up to release 14.0.0 (Rodney) is fine...starting with release 14.1.0 (Rachel), the Web-UI and console shows Senor readings as "null".
With 14.4.1 (rudolph), the sensor readings often appear (only) once, after reboot, then remain "null" after the second log entry.

As the device is located remotely, in a rural location, I have no physical access and can not check the LCD output ATM...sorry.

**Reply by Stormhand** [reply]
I have the same issue with 316D. I'll try downgrading to v.14.0

**Reply by VanillaFord** [reply]
I actually overlooked the other link given above: https://github.com/arendst/Tasmota/issues/17944 in the first place.

But if the Solution IS the individual tuning via DhtDelay, while up-/downgrade of firmware does obviously solve the issue for some installations...then the DhtDelay setting once found is not guaranteed to work with a new release.

...actually one would be in need to re-tune this from time to time / with a new upgrade.
Maybe there is a chance to have this kind of semi-automated?

@Stormhand I had the most stable results with 13.4.0...only tested with 14.0 for a couple of minutes...

**Reply by Stormhand** [reply]
Thanks @VanillaFord ill downgrade until they fix it. No idea with what DhtDelay to test.

Edit: is this possible that the rule is not working properly? I have `Rule1 on system#init do DhtDelay 500,40 endon Rule1 1` and the readings are null but when I execute `DhtDelay 500,40` manually it starts working again.

**Reply by VanillaFord** [reply]
@Stormhand ...maybe try "on System#Boot" as Trigger instead ??

**Reply by sfromis** [reply]
If you have that command as *one*, it would not work. You should enter the two `Rule1` commands separately. Do verify that the rule was stored *and* enabled as intended.

**Reply by VanillaFord** [reply]
> Rule1 on system#init do DhtDelay 500,40 endon

With that Rule1 / DhtDelay 500,40 enabled, Release 14.4.1 now works fine for me.
While Release 13.4 works reliably without tuning DhtDelay, the Console sometimes reports NULL values when the same rule is enabled. 

This confirms, that actually the need for tuning can re-occur after an update.

**Comment by bastians** [comment]
Hi, I have the same or at least a similar problem. I updated my Sonoff THR320D with THS01 to the latest Tasmota version from 12.0.0 re-configered it with the auto config option. At first I saw values briefly, but then I was still shown null values only. I have now installed version 13.4.0 and values are coming in regularly again, but I keep getting disconnections with null values, but after a short time the values come back. Before the upgrade, the connection was stable. What could be the reason for this?

**Comment by Jason2866** [comment]
Find the needed settings for your DHT sensor as described here https://github.com/arendst/Tasmota/issues/17944#issuecomment-1436657385
